### PR TITLE
supersets messages fail on convert to Pot

### PR DESCRIPTION
**Pull request by DarjaLeonova**

### CODE REVIEW

File: pkg/repo/error.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import (
+	"errors"
+
+	"golang.org/x/text/language"
+)
+
+var ErrNotFound = errors.New("entity not found")
+
+type LoadMessagesOpts struct {
```
Comment (klavinsjanis) [inline pkg/repo/error.go:11]: Why did you move LoadMessagesOpts?

**Review by kaaposc — APPROVED** [review]

File: pkg/repo/error.go (hunk 1 above)
Comment (DarjaLeonova) [reply pkg/repo/error.go:11]: Ops..
For some reason I had this file already in my local machine. mb I have created it when i was working on different ticket.
I will remove this error.go and will restore the changes inside repo.go. :)

File: pkg/convert/pot_test.go [hunk 2]
```diff
@@ -1096,6 +1096,29 @@ when * {There are {$count} apples.}
...
+							#. a greeting
+							msgid "+ {%s} hello"
+							msgstr ""
+			`),
+			expected: model.Messages{
+				Language: language.English,
+				Messages: []model.Message{
+					{
+						ID:          "+ {%s} hello",
+						Message:     "{+ {%s} hello}",
```
Comment (janishorsts) [inline pkg/convert/pot_test.go:1113]: Shouldn't we escape`{` and `}` in Message Format 2?

```suggestion
						Message:     "{+ \{%s\} hello}",
```

Comment (DarjaLeonova) [reply pkg/convert/pot_test.go]: Yes, you are right. I made changes, so now it should be correct.

File: pkg/convert/convert.go [hunk 3]
```diff
@@ -8,11 +8,49 @@ func convertToMessageFormatSingular(message string) string {
...
 	return "{" + message + "}"
 }
 
+// escapeSpecialChars escapes special characters in a string.
+func escapeSpecialChars(message string) string {
+	if strings.Contains(message, "{") ||
+		strings.Contains(message, "}") ||
+		strings.Contains(message, "\\") {
+		r := strings.NewReplacer("{", "\\{", "}", "\\}", "\\", "\\\\")
+		message = r.Replace(message)
```
Comment (janishorsts) [inline pkg/convert/convert.go:23]: Simply return it if there are no modifications.

```suggestion
		return r.Replace(message)
```

Comment (DarjaLeonova) [reply pkg/convert/convert.go:23]: Will adjust. ;)

File: pkg/convert/convert.go [hunk 4]
```diff
@@ -8,9 +8,24 @@ func convertToMessageFormatSingular(message string) string {
 		return ""
 	}
 
+	message = escapeSpecialChars(message)
+
 	return "{" + message + "}"
 }
 
+// escapeSpecialChars escapes special characters in a string.
```
Comment (janishorsts) [inline pkg/convert/convert.go:16]: can you please add a link to MF2 where special characters are defined?

File: pkg/convert/pot_test.go [hunk 5]
```diff
@@ -85,6 +85,94 @@ msgstr ""
...
+				},
+			},
+			expected: []byte(`msgid ""
+msgstr ""
+"Language: en\n"
+#. A simple greeting
+#, fuzzy
+msgid "Hello, world!"
+msgstr "Bonjour {} le monde!"
+`),
```
Comment (janishorsts) [inline pkg/convert/pot_test.go:110]: Can we reduce all test cases to what's really under the test? Fuzzy and description add noise.

```suggestion
				Messages: []model.Message{
					{
						ID:          "Hello, world!",
						Message:     `{Bonjour \{\} le monde!}`,
					},
				},
			},
			expected: []byte(`msgid ""
msgstr ""
"Language: en\n"
msgid "Hello, world!"
msgstr "Bonjour {} le monde!"
`),
```

**Review by janishorsts — APPROVED** [review]
